src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -83,10 +83,9 @@ string DefaultMessageIdStrategy(BasicDeliverEventArgs message)
             return properties.MessageId;
         }
 
-        static Dictionary<string, string> DeserializeHeaders(BasicDeliverEventArgs message)
+        static Dictionary<string, string> DeserializeHeaders(IDictionary<string, object> headers)
```

Comment:
I can't see any use of `ToDictionary()` here.

Anyhow, once this is squashed into it's a moot point.